assembly/token/impl.ts

```diff
@@ -8,6 +9,38 @@ import {ByteArray} from '@massalabs/as/assembly';
 
 const TRANSFER_EVENT_NAME = 'TRANSFER';
 const APPROVAL_EVENT_NAME = 'APPROVAL';
+const TOKEN_OWNER_STORAGE_KEY = 'token_owner_storage_key';
+
+/**
+ * Returns if the an Address is the token owner or not. Returns true if no token owner is yet set
+ * @param {Address} caller - Caller to be compared with the token owner in storage.
+ * @return {bool}
+ */
+function _assertTokenOwner(caller: Address): bool {
+  if (!Storage.has(TOKEN_OWNER_STORAGE_KEY)) {
+    return true; // no token owner set
+  }
+  // else - token owner is set, check if it equals the caller
+  const tokenOwner = Address.fromByteString(Storage.get(TOKEN_OWNER_STORAGE_KEY));
+  if (tokenOwner.equals(caller)) {
+    return true;
+  }
+
+  // token owner is not the caller
+  return false;
+}
+
+/**
+ * Sets the token owner
+ * @param {string} _args - ?
+ */
+export function setTokenOwner(_args: string): void {
+  // Check that the caller is the token owner.
+  // For initial calls, there is no owner and the check will pass.
```

Comment:
This is dangerous. your first call to setTokenOwner could be frontrunned !
i think token owner should be initialized with the token creation caller address